2_observations/src/munge_diversion_data.R

```diff
@@ -235,5 +235,45 @@ map_over_wy_files <- function(sub_dirs, root_dir) {
   return(dat2)
 }
 
+combine_release_sources <- function(out_ind, hist_rel_ind, usgs_rel_ind, modern_rel_ind, manual_rel_ind) {
+
+    hist <- readr::read_csv(sc_retrieve(hist_rel_ind, 'getters.yml'))
+    usgs <- readr::read_csv(sc_retrieve(usgs_rel_ind, 'getters.yml'))
+    modern <- readRDS(sc_retrieve(modern_rel_ind, 'getters.yml'))
+    manual <- readr::read_csv(sc_retrieve(manual_rel_ind, 'getters.yml'))
+
+    # bind together
+
+    hist_out <- group_by(hist, date, reservoir) %>%
+      summarize(release_volume_cms = sum(release_volume_cms)) %>%
+      ungroup()
+    usgs_out <- usgs %>%
+      select(reservoir, date, total_release_mgd) %>%
+      mutate(release_volume_cms = total_release_mgd*0.0438125) %>%
+      select(-total_release_mgd)
+    manual_out <- manual %>%
+      pivot_longer(cols = -Date, names_to = 'reservoir', values_to = 'release_volume_cfs') %>%
+      mutate(release_volume_cms = release_volume_cfs/35.314666) %>%
+      mutate(date = as.Date(Date, format = '%m/%d/%Y')) %>%
+      select(-release_volume_cfs, -Date) %>%
+      filter(!is.na(release_volume_cms))
+
+    # group by and slice_min picks the sources in order of the bind
+    # e.g., always use the historical data from the NYC DEP, and then prioritize
+    # data in the order of modern pull from NWIS, manually data from ODRM,
+    # and data imported from the NY WSC (NY WSC data was messies/most prone to
+    # error in importing)
+    all <- bind_rows(hist_out, modern, manual_out, usgs_out, .id = 'id') %>%
+      group_by(reservoir, date) %>%
+      slice_min(id) %>%
```

Comment:
I think `slice_min` is probably operating alphabetically here on `id`, not by order?  Same effect since hist_out is still first, just FYI. 

```
> one <- starwars[1:4, ]
> two <- starwars[9:12, ]
> bind_rows(list(a = one, b = two), .id = "id")
# A tibble: 8 x 15
  id    name  height  mass hair_color skin_color eye_color birth_year sex   gender homeworld species films vehicles starships
  <chr> <chr>  <int> <dbl> <chr>      <chr>      <chr>          <dbl> <chr> <chr>  <chr>     <chr>   <lis> <list>   <list>   
1 a     Luke…    172    77 blond      fair       blue            19   male  mascu… Tatooine  Human   <chr… <chr [2… <chr [2]>
2 a     C-3PO    167    75 NA         gold       yellow         112   none  mascu… Tatooine  Droid   <chr… <chr [0… <chr [0]>
3 a     R2-D2     96    32 NA         white, bl… red             33   none  mascu… Naboo     Droid   <chr… <chr [0… <chr [0]>
4 a     Dart…    202   136 none       white      yellow          41.9 male  mascu… Tatooine  Human   <chr… <chr [0… <chr [1]>
5 b     Bigg…    183    84 black      light      brown           24   male  mascu… Tatooine  Human   <chr… <chr [0… <chr [1]>
6 b     Obi-…    182    77 auburn, w… fair       blue-gray       57   male  mascu… Stewjon   Human   <chr… <chr [1… <chr [5]>
7 b     Anak…    188    84 blond      fair       blue            41.9 male  mascu… Tatooine  Human   <chr… <chr [2… <chr [3]>
8 b     Wilh…    180    NA auburn, g… fair       blue            64   male  mascu… Eriadu    Human   <chr… <chr [0… <chr [0]>
> bind_rows(list(a = one, b = two), .id = "id") %>% slice_min(name)
# A tibble: 1 x 15
  id    name  height  mass hair_color skin_color eye_color birth_year sex   gender homeworld species films vehicles starships
  <chr> <chr>  <int> <dbl> <chr>      <chr>      <chr>          <dbl> <chr> <chr>  <chr>     <chr>   <lis> <list>   <list>   
1 b     Anak…    188    84 blond      fair       blue            41.9 male  mascu… Tatooine  Human   <chr… <chr [2… <chr [3]>
```

Reply:
When the list isn't named (similar to how I used it), the argument `.id` uses number in the order the objects are provided, e.g.:

```
> a <- tibble(x = 1:4)
> b <- tibble(x = 5:10)
> bind_rows(b, a, .id = 'id')
# A tibble: 10 x 2
   id        x
   <chr> <int>
 1 1         5
 2 1         6
 3 1         7
 4 1         8
 5 1         9
 6 1        10
 7 2         1
 8 2         2
 9 2         3
10 2         4
```
which is a nice shorthand for explicitly choosing based on provided importance (order).